src/qtsoap.h

```diff
@@ -274,12 +192,12 @@ class QT_QTSOAP_EXPORT QtSoapArray : public QtSoapType
     const QtSoapType &operator [](const QString &) const;
     const QtSoapType &operator [](const QtSoapQName &) const;
 
-    void append(QtSoapType *item);
-    void insert(int pos0, QtSoapType *item);
-    void insert(int pos0,int pos1, QtSoapType *item);
-    void insert(int pos0,int pos1,int pos2, QtSoapType *item);
-    void insert(int pos0,int pos1,int pos2,int pos3, QtSoapType *item);
-    void insert(int pos0,int pos1,int pos2,int pos3,int pos4, QtSoapType *item);
+    void append(std::shared_ptr<QtSoapType> item);
+    void insert(int pos0, std::shared_ptr<QtSoapType> item);
```

Comment:
I was going to :trollface: you on the line breaks...than I thought it would have been too much for your heart, on a monday.

Reply:
Thought about formatting too (because we know that formatting is an hot topic here). Formatting has been done automagically by clang-format according to these rules: https://github.com/Proemion/QtSOAP/blob/shared_pointers/.clang-format

Reply:
Well, I consider that as a troll because that's not really one of our libs so I don't think it should be formatted according to our guidelines. At least, we didn't reformat gpsd or the qml proxy one.
(But the rules are cool!)